### PR TITLE
[DOC] clarify the config for mounting secrets as volumes

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

Added some additional information to the _Mounting Secrets as volumes for connector configuration_ in the _Using Guide_ to make it clearer how config providers are used.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by PaulRMellor** [comment]
This PR fixes https://github.com/strimzi/strimzi-kafka-operator/issues/2452

File: documentation/modules/proc-kafka-connect-mounting-volumes.adoc [hunk 1]
```diff
@@ -43,42 +55,48 @@ metadata:
 spec:
   # ...
   config:
-    config.providers: file
-    config.providers.file.class: org.apache.kafka.common.config.provider.FileConfigProvider
+    config.providers: file <1>
```
Comment (ppatierno) [inline documentation/modules/proc-kafka-connect-mounting-volumes.adoc:58]: is there a kind of list of supported config providers somewhere @tombentley ?

File: documentation/modules/proc-kafka-connect-mounting-volumes.adoc [hunk 2]
```diff
@@ -43,42 +55,48 @@ metadata:
...
   #...
   externalConfiguration:
     volumes:
-      - name: connector-config
+      - name: connector-config <3>
         secret:
-          secretName: mysecret
+          secretName: mysecret <4>
 ----
+<1> The name of the configuration provider. Use a comma-separted list for more than one provider.
```
Comment (tombentley) [inline documentation/modules/proc-kafka-connect-mounting-volumes.adoc:67]: Right now it's only really possible to use the `FileConfigProvider`, but let's keep the 2nd sentence in anticipation of  KIP-632.

File: documentation/modules/proc-kafka-connect-mounting-volumes.adoc [hunk 3]
```diff
@@ -43,42 +55,48 @@ metadata:
...
+<2> The `FileConfigProvider` class used by the configuration provider to access the properties file.
```
Comment (tombentley) [inline documentation/modules/proc-kafka-connect-mounting-volumes.adoc:68]: It's not _used by_ the provider, it _is_ the provider.

```suggestion
<2> The `FileConfigProvider` is a configuration provider which provides values from properties files.
```

File: documentation/modules/proc-kafka-connect-mounting-volumes.adoc [hunk 4]
```diff
@@ -43,42 +55,48 @@ metadata:
...
       "database": "my-postgresql:5432",
-      "username":"${file:/opt/kafka/external-configuration/connector-config/connector.properties:dbUsername}",
-      "password":"${file:/opt/kafka/external-configuration/connector-config/connector.properties:dbPassword}",
+      "username":"${file:/opt/kafka/external-configuration/connector-config/connector.properties:dbUsername}", <1>
+      "password":"${file:/opt/kafka/external-configuration/connector-config/connector.properties:dbPassword}", <2>
       # ...
    }
 }
 ----
+<1> Path to the username property value. The path takes the form `/opt/kafka/external-configuration/_SECRET-VOLUME-NAME_/_PROPERTIES-FILENAME:USERNAME-VALUE_`.
```
Comment (tombentley) [inline documentation/modules/proc-kafka-connect-mounting-volumes.adoc:95]:
```suggestion
<1> Path to the username property value. The path takes the form `/opt/kafka/external-configuration/_SECRET-VOLUME-NAME_/_PROPERTIES-FILENAME:USERNAME-PROPERTY_`.
```

File: documentation/modules/proc-kafka-connect-mounting-volumes.adoc [hunk 5]
```diff
@@ -43,42 +55,48 @@ metadata:
...
+<2> Path to the password property value. The path takes the form `/opt/kafka/external-configuration/_SECRET-VOLUME-NAME_/_PROPERTIES-FILENAME:PASSWORD-VALUE_`.
```
Comment (tombentley) [inline documentation/modules/proc-kafka-connect-mounting-volumes.adoc:96]:
```suggestion
<2> Path to the password property value. The path takes the form `/opt/kafka/external-configuration/_SECRET-VOLUME-NAME_/_PROPERTIES-FILENAME:PASSWORD-PROPERTY_`.
```

File: documentation/modules/proc-kafka-connect-mounting-volumes.adoc (hunk 1 above)
Comment (ppatierno) [reply documentation/modules/proc-kafka-connect-mounting-volumes.adoc:58]: So I think that we should explain that in the examples as you highlight that `config.providers` defines an alias and you have to use that alias in the `config.proviers.${alias}.class` parameter. Tbh it was not clear to me.

File: documentation/modules/proc-kafka-connect-mounting-volumes.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/modules/proc-kafka-connect-mounting-volumes.adoc:67]: Thanks for clarifying Tom. I changed to:

<1> The alias for the configuration provider. Use a comma-separated list if you want to add more than one provider.

Let me know if you think we need to add more.

File: documentation/modules/proc-kafka-connect-mounting-volumes.adoc (hunk 1 above)
Comment (ppatierno) [reply documentation/modules/proc-kafka-connect-mounting-volumes.adoc:58]: @PaulRMellor ^^

Comment (PaulRMellor) [reply documentation/modules/proc-kafka-connect-mounting-volumes.adoc:58]: @ppatierno - how about:

> <1> The alias for the configuration provider, which is used to define other configuration properties. 
> Use a comma-separated list if you want to add more than one provider.
> <2> The `FileConfigProvider` is the configuration provider that provides values from properties files.
> The property uses the alias from `config.providers`, taking the form `config.providers.${alias}.class`.

File: documentation/modules/proc-kafka-connect-mounting-volumes.adoc [hunk 6]
```diff
@@ -25,13 +33,17 @@ metadata:
...
+    dbUsername: my-user <2>
     dbPassword: my-password
 ----
+<1> The connector configuration in properties file format.
+<2> Database _username_ and _password_ properties used in the configuration.
 
 . Create or edit the Kafka Connect resource.
-Configure the `FileConfigProvider` in the `config` section and the `externalConfiguration` section of the `KafkaConnect` or `KafkaConnectS2I` custom resource to reference the secret.
++
+Configure the configuration provider in the `config` section and the `externalConfiguration` section of the `KafkaConnect` or `KafkaConnectS2I` custom resource to reference the secret.
```
Comment (laidan6000) [inline documentation/modules/proc-kafka-connect-mounting-volumes.adoc:45]: This sentence is quite tdifficult to parse! You could change it as follows:

In the `config` section and the `externalConfiguration` section of the `KafkaConnect` or `KafkaConnectS2I` custom resource, configure the configuration provider to reference the secret.

File: documentation/modules/proc-kafka-connect-mounting-volumes.adoc [hunk 7]
```diff
@@ -25,13 +33,17 @@ metadata:
...
 type: Opaque
 stringData:
-  connector.properties: |-
-    dbUsername: my-user
+  connector.properties: |- <1>
+    dbUsername: my-user <2>
     dbPassword: my-password
 ----
+<1> The connector configuration in properties file format.
+<2> Database _username_ and _password_ properties used in the configuration.
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/modules/proc-kafka-connect-mounting-volumes.adoc:41]: Not really. Removed.

**Review by laidan6000 — APPROVED** [review]
Changes look good.

**Pull request merged**